src/selections/parser.cpp

```diff
@@ -13,93 +14,145 @@
+    {"rad2deg", [](MathAst ast){ return chemfiles::make_unique<Function>([](double rad){ return rad * 180 / PI; }, "rad2deg", std::move(ast));}},
+    {"deg2rad", [](MathAst ast){ return chemfiles::make_unique<Function>([](double deg){ return deg * PI / 180; }, "deg2rad", std::move(ast));}},
 };
 
-using numeric_variable_functions_creator_t = std::function<MathAst(std::vector<Variable>)>;
+using numeric_variable_functions_creator_t = std::function<MathAst(SelectionArguments)>;
 struct NumericVariableFunction {  // NOLINT: constructor does not initialize these fields: arity
     unsigned arity;
     numeric_variable_functions_creator_t creator;
 };
 
 static std::map<std::string, NumericVariableFunction> NUMERIC_VAR_FUNCTIONS = {
-    {"distance", {2, [](std::vector<Variable> args){ return MathAst(new Distance(args[0], args[1])); }}},
-    {"angle", {3, [](std::vector<Variable> args){ return MathAst(new Angle(args[0], args[1], args[2])); }}},
-    {"dihedral", {4, [](std::vector<Variable> args){ return MathAst(new Dihedral(args[0], args[1], args[2], args[3])); }}},
-    {"out_of_plane", {4, [](std::vector<Variable> args){ return MathAst(new OutOfPlane(args[0], args[1], args[2], args[3])); }}},
+    {"distance", {2, [](SelectionArguments args) {
+        assert(args.count == 2);
+        return make_unique<Distance>(
+            std::move(args.values[0]),
+            std::move(args.values[1])
+        );
+    }}},
+    {"angle", {3, [](SelectionArguments args) {
+        assert(args.count == 3);
+        return make_unique<Angle>(
+            std::move(args.values[0]),
+            std::move(args.values[1]),
+            std::move(args.values[2])
+        );
+    }}},
+    {"dihedral", {4, [](SelectionArguments args) {
+        assert(args.count == 4);
+        return make_unique<Dihedral>(
+            std::move(args.values[0]),
+            std::move(args.values[1]),
+            std::move(args.values[2]),
+            std::move(args.values[3])
+        );
+    }}},
+    {"out_of_plane", {4, [](SelectionArguments args) {
+        assert(args.count == 4);
+        return make_unique<OutOfPlane>(
+            std::move(args.values[0]),
+            std::move(args.values[1]),
+            std::move(args.values[2]),
+            std::move(args.values[3])
+        );
+    }}},
 };
 
 
-using bool_selector_creator_t = std::function<Ast(std::vector<SubSelection>)>;
+using bool_selector_creator_t = std::function<Ast(SelectionArguments)>;
 struct BooleanFunction {  // NOLINT: constructor does not initialize these fields: arity
     unsigned arity;
     bool_selector_creator_t creator;
 };
 
 static std::map<std::string, BooleanFunction> BOOLEAN_SELECTORS = {
-    {"all", {0, [](std::vector<SubSelection> /*unused*/) {  // NOLINT: can not make std::vector<SubSelection> a const reference
+    {"all", {0, [](SelectionArguments /*unused*/) {  // NOLINT: can not make std::vector<SubSelection> a const reference
         return Ast(new All());
```

Comment:
Is there any reason why we don't use `chemfiles::make_unique<All>()` here?
This applies for the whole map.

Reply:
No reason, and it make sense to use it. I'll change this!